src/lib/mina/account-update.unit-test.ts

```diff
@@ -122,3 +131,40 @@ function createAccountUpdate() {
     'Check signature: Invalid signature on fee payer for key'
   );
 }
+
+// throws an error when both flags are true in check method
+{
+  expect(() => {
+    AccountUpdate.MayUseToken.type.check({
+      parentsOwnToken: Bool(true),
+      inheritFromParent: Bool(true),
+    });
+  }).toThrowError(
+    'MayUseToken: parentsOwnToken and inheritFromParent cannot both be true'
+  );
+}
+
+// throws an error when both flags are true in AccountUpdate.check method
+// TODO: This should be throwing but it isn't, need to investigate
```

Comment:
Under the assumption of this [comment](https://github.com/o1-labs/o1js/pull/1716#discussion_r1673801107), I would suspect that `AccountUpdate.check` should throw, but it isn't here :thinking:

Reply:
aren't you missing the Mina changes that you had in the other PR?

Reply:
Ah, I didn't know that was necessary. So for an `AccountUpdate` to recursively call `MayUseToken.check`, we need this [change](https://github.com/MinaProtocol/mina/pull/15808/files#diff-1edc359c0b2f3f296bb16dd374c402ecfb87b2de6d1f0d28968966c26e61fd2bR1141)?

Reply:
yeah exactly

Reply:
the info goes into `js-layout.ts`